src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,356 @@
+
+      opacity.value = VISIBLE
+
+      // When sliding the thumb across a distance shorter than the track's width
+      if (progressing < MINIMUM_TRACK_WIDTH) {
+        runOnJS(updateSlider)(MINIMUM_TRACK_WIDTH, minimumValue)
+      }
+      // When sliding the thumb over the track's width
+      else if (progressing > trackWidth) {
+        runOnJS(updateSlider)(trackWidth, maximumValue)
+      } else {
+        const value = progressing / (trackWidth / (maximumValue - minimumValue))
+        runOnJS(updateSlider)(progressing, value + minimumValue)
+      }
+    },
+    onEnd: () => {
+      const {range} = sliderInfo.value
+      let value = roundToValue ? sliderValue.value.toFixed(roundToValue) : sliderValue.value
+      opacity.value = INVISIBLE
+      if (step) {
+        // Calculate the ratio of the current thumb position with respect to the entire range of the slider
+        const progressRatio = Math.round(progress.value / range)
+        value = minimumValue + progressRatio * stepValue
+        const roundedProgress = progressRatio * range
+        runOnJS(updateSlider)(roundedProgress, value)
+      }
+      runOnJS(onValueChange)(value as number)
+    },
+  })
+
+  /**
+   * Update the tracked width based on the thumb sliding
+   */
+  const animatedTrackStyle = useAnimatedStyle(
+    () => ({
+      width: withTiming(progress.value, {duration: 1}),
+    }),
+    [progress],
+  )
+
+  /**
+   * Add animation to the thumb while it's sliding
+   */
+  const animatedThumbStyle = useAnimatedStyle(
+    () => ({
+      transform: [{translateX: progress.value}],
+    }),
+    [progress],
+  )
+
+  /**
+   * Update opacity when touching the thumb
+   * This animation should run within 200ms when the thumb is touched or released
+   */
+  const opacityStyle = useAnimatedStyle(
+    () => ({
+      opacity: withTiming(opacity.value, {
+        duration: DURATION,
+      }),
+    }),
+    [opacity],
+  )
+
+  /**
+   * Display the current slider value in real time as the user slides the thumb along the slider track
+   */
+  const animatedProps = useAnimatedProps(
+    () =>
+      ({
+        text: `${roundToValue !== undefined ? sliderValue.value.toFixed(roundToValue) : sliderValue.value}`,
+      } as AnimatedLabelProps),
+  )
+
+  const getTrackWidth = (event: LayoutChangeEvent) => {
+    const {width} = event.nativeEvent.layout
+    // Range refers to the width of a point
+    // It is used to calculate the correct position of the slider while sliding
+    const range = width / totalPoint
+    sliderInfo.value = {range: range, trackWidth: width}
+  }
+
+  /**
+   * Function called when the user presses on a point on the slider's track
+   * Calculates the position of the pressed point on the slider's track and updates the slider's state accordingly
+   * @param {number} point - The index of the point on the slider's track that was pressed by the user
+   */
+  const onPressPoint = useCallback(
+    (point: number) => {
```

Comment:
recommend change to pointIndex or index for matching meaning. This named **point** sounds like it is the value not the index

Reply:
updated